### DISCUSSION TITLE
Help Picking GoPro Model for Long-Form Timelapse in Unique Conditions

### DISCUSSION THREAD

**Discussion by simon2071** [gopro/labs#851] [Q&A]
Hello! 

I am new to the GoPro Labs software and this entire discussion forum in general, but am hoping you all could help me decide if a GoPro can satisfy my requirements for an upcoming project (and potentially which model would work best as well). 

I am an aquatic biologist and am currently working on developing a long-term monitoring program for a benthic invertebrate that exists in the ecosystem in which I work. The project is unique in that the conditions of the lake are often quite extreme, with both temperature and salinity fluctuating greatly throughout the year. My plan is to build a custom, waterproof housing out of PVC and glass that will hold my camera of choice. This unit will then be anchored at the bottom of the lake for at least one month. During that time, I want a camera that will take photographs ideally three times per day (or at the very least once per day) of the exact same location. Due to logistical constraints, I will not be able to retrieve the camera or even service it more than once per month. Additionally, I plan to place a simple computer inside the housing (I am thinking Raspberry Pi to keep costs low and minimize the amount of code I have to troubleshoot?). This computer will be used to program an LED light fixed to the outside of the housing and a miniature windshield wiper, both of which will activate immediately prior to the camera capturing its daily photograph. For reference, the LED will provide much needed light at the bottom of the lake (which is quite turbid and dark) while the mini wiper will clear algal growth from the exterior of the housing's glass section prior to the camera taking any photos. 

After all of that background information, here is what I am envisioning I need:
-- Camera with thermal tolerance as low as -5C that can be programmed to take at least 1 photo per day _at exactly the same time_ for >1 month
-- Minimum 1080p photographs (better visual fidelity is desirable as the low light and turbid waters make photographing the sample area difficult)
-- The battery life of the camera must last >30 days _without_ external batteries, as I cannot retrieve the camera during the month to replace/charge the unit. I did read on this forum that there are continuous portable battery chargers that can power a GoPro via solar energy, but being at the bottom of the lake renders this option impossible.
-- Due to a restricted budget, the cheaper I can make this entire contraption the better! Furthermore, I would eventually like to build several of these units and place them around the lake, making it even more important to be cost effective. 

To reduce the amount of custom programming/debugging I must do, and because of the various models' tolerances of extreme conditions, I am thinking GoPro (with Labs) may be the best options for this project. But I also am aware I am asking a lot of the camera. What are your opinions? I'm also open to any advice and am happy to provide more details as to the conditions in the lake/camera requirements!

Thanks in advance for any help you all may provide, it is greatly appreciated.

**Comment by dnewman-gpsw** [comment]
This is totally doable, you get about 40-50 days of operation at three photos a day using HERO12.  I haven't run tests at -5C but the battery is rated to -10C.

**Reply by dnewman-gpsw** [reply]
HERO11 is also an option, but the reliability for this improved with later models.

**Reply by simon2071** [reply]
That is fantastic! 

HERO 12 definitely seems the best option. I have been digging through the forums and it appears as though there are several ways that I see people coding for long-form timelapse projects. Is the code model-specific, or will I be able to use any line of code I find through the forum that people report works with the HERO 12?

I am also new to GitHub and may have missed it somewhere, but is there a link to the syntax used to code in GoPro Lab? This is a new language to me.

I am extremely grateful for the advice!

**Reply by dnewman-gpsw** [reply]
Yes HERO12 pretty runs all the commands on the forum.  The command language for HERO10/11/12/Mini and somewhat MAX are all the same, older cameras don't have advanced scripting, but the settings are mostly the same.  

The language was developed to pack a lot into QR codes, so it is "new".  For some documentation start here: https://gopro.github.io/labs/control/tech/

Also you can learn a lot from using https://gopro.github.io/labs/build/

**Reply by simon2071** [reply]
Thanks so much! I am going to keep brainstorming for the rest of the device's components but will move forward with the HERO 12 as my camera of choice. Your input was incredibly helpful!

**Comment by caden-fluegge** [comment]
Hi!  I am looking to do a similar project but with video in a pelagic setting.  We are trying to figure out a way around biofouling and the option of a wiper came to mind - are you making a custom one or using a pre-made option?

**Reply by simon2071** [reply]
Hello! So that is currently up for debate. I recently found an option for a patented version of a programmable wiper, named the miniWiper and manufactured by Precision Measurement Engineering. But that little gadget runs ~$1,000 and only cleans a few centimeters of glass. We are in the process of brainstorming a custom, programmable wiper (potentially from a motorcycle helmet) that would be fixed to our housing and attached to some type of simple computer (Raspberry Pi?) secured inside the housing. But we are open to any other ideas, if your team has had some success with an alternative method!
